src/Apps/Consign/Routes/ConsignmentInquiry/ConsignmentInquiry.tsx

```diff
@@ -0,0 +1,209 @@
+import { useState } from "react"
+import { TopContextBar } from "Components/TopContextBar"
+
+const logger = createLogger("ConsignmentInquiry/ConsignmentInquiry.tsx")
+
+const getContactInformationFormInitialValues = (
+  me: ConsignmentInquiry_me$data
+): ConsignmentInquiryFormModel => ({
+  name: me?.name || "",
+  email: me?.email || "",
+  phoneNumber:
+    me?.phoneNumber?.national ||
+    me?.phone ||
+    me?.phoneNumber?.international ||
+    "",
+  phoneNumberCountryCode: me?.phoneNumber?.regionCode || "us",
+  message: "",
+})
+
+export interface ConsignmentInquiryProps {
+  me: ConsignmentInquiry_me$data
+}
+
+export const ConsignmentInquiry: React.FC<ConsignmentInquiryProps> = ({
+  me,
+}) => {
+  const [showModal, setShowModal] = useState(false)
+  const { trackEvent } = useTracking()
+  const { router } = useRouter()
+  const { sendToast } = useToasts()
+  const { relayEnvironment } = useSystemContext()
+  const initialValue = getContactInformationFormInitialValues(me)
+  const initialErrors = validate(
+    initialValue,
+    consignmentInquiryValidationSchema
+  )
+
+  const handleRecaptcha = (action: RecaptchaAction) =>
+    new Promise(resolve => recaptcha(action, resolve))
+
+  const handleSubmit = async ({
+    name,
+    email,
+    phoneNumber,
+    phoneNumberCountryCode,
+    message,
+  }: ConsignmentInquiryFormModel) => {
+    if (!(await handleRecaptcha("consignment_inquiry"))) {
+      return
+    }
+
+    let phoneNumberInternational: string | undefined = undefined
+    if (phoneNumber.trim().length) {
+      phoneNumberInternational = `+${
+        COUNTRY_CODES[phoneNumberCountryCode.toLocaleUpperCase()]
+      } ${phoneNumber.trim()}`
+    }
+
+    const input: CreateConsignmentInquiryMutationInput = {
+      email,
+      message,
+      name,
+      phoneNumber: phoneNumberInternational,
+      userId: me?.internalID,
+    }
+
+    if (relayEnvironment) {
+      try {
+        const consignmentInquiryId = await createConsignmentInquiry(
+          relayEnvironment,
+          input
+        )
+        if (consignmentInquiryId) {
+          trackEvent(tracks.sentConsignmentInquiry(consignmentInquiryId))
+          router.replace({ pathname: "/sell/inquiry" })
+          router.push("/sell/inquiry/sent")
+        }
+      } catch (error) {
+        logger.error(error)
+        sendToast({
+          variant: "error",
+          message: `Error: ${error}`,
+          description: "Please contact sell@artsy.net",
+        })
+      }
+    }
+  }
+
+  const handleBack = (isDirty: boolean) => {
+    if (isDirty) {
+      setShowModal(true)
+      return
+    }
+    router.go(-1)
+  }
+
+  const handleCloseModal = () => {
+    // Every time the modal shows up, we need to decrement by -1, else
+    // router needs to pop as many times as the number of times the modal
+    // shows up in order to correctly pop to the sell with artsy screen.
+    router.go(-1)
```

Comment:
**suggestion:** Let's check this carefully with a few browsers. We had some issues with using `router.go(-1)` in the My Collection artwork upload flow.

Reply:
@olerichter00  You are correct. go(-1) behaves weirdly on Safari. It works fine when triggered from Buttons. But fails when used with RouterLink.